liquidctl/driver/msi.py

```diff
@@ -0,0 +1,40 @@
+"""liquidctl drivers for MSI liquid coolers.
+
+Supported devices:
+
+- MPG Coreliquid K360
+
+Copyright (C) 2021  Andrew Udvare and contributors
+SPDX-License-Identifier: GPL-3.0-or-later
+"""
+
+from liquidctl.driver.usb import UsbHidDriver
+
+_REPORT_LENGTH = 64
+
+
+class MpgCooler(UsbHidDriver):
+    SUPPORTED_DEVICES = [
+        (0x0db0, 0xb130, None, 'MSI MPG Coreliquid K360', {}),
+    ]
+
+    def get_status(self, **kwargs):
+        buf = bytearray(_REPORT_LENGTH)
+        buf[1] = 0xd0
+        buf[2] = 0x31
+        self.device.clear_enqueued_reports()
+        self.device.write(buf)
+        array = bytes(self.device.read(_REPORT_LENGTH))
+        assert array[1] == 0x31, 'Unexpected value in response buffer'
```

Comment:
Is there a reason for this assertion? Are there other possible values that could be here that represent a known error case?

Reply:
This check is in the original code that MSI wrote so I thought I should replicate it.

Reply:
> Are you able to also add a protocol description to the `docs/developer/protocol` folder and open a related PR to https://github.com/liquidctl/collected-device-data that adds some packet traces.

Yes. I can provide [a lot](https://github.com/Tatsh/liquidctl/commit/390baa396cbb2b610cc7090ef80b98b734bd2571).

Unfortunately the device I got was defective and I already sent it back to Newegg for a refund. The branch linked has every single function converted from the C# decompilation to Python. I have tested everything with the real hardware except uploading image files to the OLED.

I went with a Corsair device that has a Commander Core, which is not fully implemented or bug free. In particular, every time `status` is called it is resetting the device's power state causing it to blink. The iCUE software does not do this.

Reply:
Is there a reason why you are not making the PR using [this branch](https://github.com/Tatsh/liquidctl/commit/390baa396cbb2b610cc7090ef80b98b734bd2571)? if everything was tested there to see if it worked?

Also the protocol description would be something like [this](https://github.com/liquidctl/liquidctl/blob/main/docs/developer/protocol/lighting_node_rgb.md) rather than an implementation, it allows for easier understanding of the protocol than reading through the code, and the protocol _should_ remain consistent even if the actual implementation changes or is only implemented in parts (as is the case for this PR)